src/NHibernate/Loader/Loader.cs

```diff
@@ -1708,60 +1716,54 @@ private CacheableResultTransformer CreateCacheableResultTransformer(QueryParamet
 				queryParameters.HasAutoDiscoverScalarTypes, SqlString);
 		}
 
-		internal IList GetResultFromQueryCache(ISessionImplementor session, QueryParameters queryParameters,
-											  ISet<string> querySpaces, IQueryCache queryCache,
-											  QueryKey key)
+		internal bool CanGetFromCache(ISessionImplementor session, QueryParameters queryParameters)
 		{
-			IList result = null;
-
-			if (!queryParameters.ForceCacheRefresh && session.CacheMode.HasFlag(CacheMode.Get))
-			{
-				IPersistenceContext persistenceContext = session.PersistenceContext;
+			return !queryParameters.ForceCacheRefresh && session.CacheMode.HasFlag(CacheMode.Get);
+		}
 
-				bool defaultReadOnlyOrig = persistenceContext.DefaultReadOnly;
+		internal IList GetResultFromQueryCache(
+			ISessionImplementor session, QueryParameters queryParameters, ISet<string> querySpaces,
+			IQueryCache queryCache, QueryKey key)
+		{
+			if (!CanGetFromCache(session, queryParameters))
+				return null;
 
-				if (queryParameters.IsReadOnlyInitialized)
-					persistenceContext.DefaultReadOnly = queryParameters.ReadOnly;
-				else
-					queryParameters.ReadOnly = persistenceContext.DefaultReadOnly;
```

Comment:
This handling has been moved on the standard query cache side, as written in previous comments.